### ISSUE TITLE
Subtractive drawing mode

### ISSUE THREAD

**Issue by KoBeWi** [gosu/gosu#413]
Right now, Gosu supports _:additive_ and _:multiply_ drawing modes. I've been using them extensively recently and I realized something like _:subtractive_ (or _:subtract_) mode, which subtracts color values, would be useful too.

**Comment by jlnr** [comment]
Absolutely. I think it would be even cooler to have the full suite of Photoshop blend modes available (this is from the PDF file format reference, but I think these are the same as in Photoshop):

https://www.adobe.com/content/dam/Adobe/en/devnet/pdf/pdfs/pdf_reference_archives/blend_modes.pdf

I'm not sure how many of them can be implemented with just plain OpenGL; maybe some of them need shaders.
